### DISCUSSION TITLE
Gradle wrappers taking GBs of space

### DISCUSSION THREAD

**Discussion by pbetkier** [renovatebot/renovate#8108] [Request Help] [answered]
We run self-hosted Renovate on 300+ repositories, mostly using Gradle. Our Renovate runs on CI fail to update some repositories due to disk space errors.

By the end of the run, our Renovate container has 8G under `/home/ubuntu/.gradle/wrapper/dists/` and 2.5G under `/home/ubuntu/.gradle/caches/`:

```
$ du -h -d1 /home/ubuntu/.gradle/wrapper/dists/
209M	/home/ubuntu/.gradle/wrapper/dists/gradle-6.4.1-bin
423M	/home/ubuntu/.gradle/wrapper/dists/gradle-6.7.1-all
208M	/home/ubuntu/.gradle/wrapper/dists/gradle-6.3-bin
311M	/home/ubuntu/.gradle/wrapper/dists/gradle-4.5.1-all
420M	/home/ubuntu/.gradle/wrapper/dists/gradle-6.3-all
418M	/home/ubuntu/.gradle/wrapper/dists/gradle-6.6-all
418M	/home/ubuntu/.gradle/wrapper/dists/gradle-6.6.1-all
210M	/home/ubuntu/.gradle/wrapper/dists/gradle-6.7.1-bin
180M	/home/ubuntu/.gradle/wrapper/dists/gradle-5.2.1-bin
210M	/home/ubuntu/.gradle/wrapper/dists/gradle-6.6.1-bin
403M	/home/ubuntu/.gradle/wrapper/dists/gradle-6.1.1-all
209M	/home/ubuntu/.gradle/wrapper/dists/gradle-6.5.1-bin
422M	/home/ubuntu/.gradle/wrapper/dists/gradle-6.4.1-all
192M	/home/ubuntu/.gradle/wrapper/dists/gradle-5.6.2-bin
210M	/home/ubuntu/.gradle/wrapper/dists/gradle-6.6-bin
422M	/home/ubuntu/.gradle/wrapper/dists/gradle-6.5.1-all
209M	/home/ubuntu/.gradle/wrapper/dists/gradle-6.5-bin
210M	/home/ubuntu/.gradle/wrapper/dists/gradle-6.7-bin
440M	/home/ubuntu/.gradle/wrapper/dists/gradle-6.0.1-all
179M	/home/ubuntu/.gradle/wrapper/dists/gradle-5.4.1-bin
403M	/home/ubuntu/.gradle/wrapper/dists/gradle-6.1-all
437M	/home/ubuntu/.gradle/wrapper/dists/gradle-5.6.3-all
176M	/home/ubuntu/.gradle/wrapper/dists/gradle-5.0-bin
195M	/home/ubuntu/.gradle/wrapper/dists/gradle-6.0-bin
874M	/home/ubuntu/.gradle/wrapper/dists/gradle-5.6.4-all
423M	/home/ubuntu/.gradle/wrapper/dists/gradle-6.7-all
192M	/home/ubuntu/.gradle/wrapper/dists/gradle-5.6.3-bin
8.4G	/home/ubuntu/.gradle/wrapper/dists/
```

What's the solution? Do we need to give ~20G of disk space to our Renovate container? Or should Renovate clear Gradle wrappers during execution once they take too much space? Maybe the experimental `gradle-lite` manager is the solution?

**Comment by viceice** [comment, marked as answer]
This is a gradle issue, we can't really do much here. You can test `gradle-lite`  or try the slim renovate, it will run gradle in side container, which will be deleted after one-time usage.

**Reply by rarkins** [reply]
If you're a heavy gradle user then helping us find any gaps in what we've called "gradle-lite" would be great. I think you'll find it runs a lot faster too, e.g. saving minutes per repository.

**Reply by pbetkier** [reply]
Thanks, I'll give `gradle-lite` a try then. I'd love to compare dry-run reports for `gradle` and `gradle-lite` before enabling it in our ecosystem. Is there a way to do it easily? Comparing raw logs would likely give a lot of noise.

**Reply by viceice** [reply]
you can check the `packageFiles with updates` debug message, there is a json dump after it. there you can compare the `gradle` and `gradle-lite` objects
